datastore/longhorn.go

```diff
@@ -3702,7 +3702,7 @@ func (s *DataStore) DeleteRecurringJob(name string) error {
 }
 
 func ValidateRecurringJob(job longhorn.RecurringJobSpec) error {
-	if job.Cron == "" || job.Task == "" || job.Name == "" || job.Retain == 0 {
+	if job.Cron == "" || job.Task == "" || job.Name == "" {
```

Comment:
`job.Retain < 0`

Reply:
We have the mutation for `recurringjob.Spec.Retain`, do we still need this check?

Reply:
I mean, we can allow `retain == 0`, right? If the type is `snapshot-delete`, `retain == 0` means retaining the volume head only.

Reply:
Yes, so removing it from the validation here. And I think we don't need to validate `job.Retain < 0` since it's included in the mutation.